src/pg/pg-store.ts

```diff
@@ -621,6 +630,36 @@ export class PgStore extends BasePgStore {
     return inscription_id;
   }
 
+  private async insertInscriptionCount(args: { block_height: number }): Promise<void> {
+    await this.sql`
+      WITH previous AS (
+        SELECT COALESCE((
+          SELECT inscriptions_total
+          FROM inscriptions_per_block
+          WHERE block_height = ${args.block_height - 1}
+        ), 0) as previous_total
+      ), current AS (
+        SELECT
+          block_height,
+          COUNT(*) as inscriptions,
+          COUNT(*) + (SELECT previous_total FROM previous) as inscriptions_total
+        FROM locations
+        WHERE block_height = ${args.block_height} AND genesis = true
+        GROUP BY block_height
+        LIMIT 1
+      )
+      INSERT INTO inscriptions_per_block
+      SELECT * FROM current
+      ON CONFLICT (block_height) DO UPDATE SET
+        inscriptions = EXCLUDED.inscriptions,
+        inscriptions_total = EXCLUDED.inscriptions_total;
+    `;
+  }
+
+  private async rollBackInscriptionCount(args: { block_height: number }): Promise<void> {
+    await this.sql`DELETE FROM inscriptions_per_block WHERE block_height = ${args.block_height}`;
```

Comment:
Went with something similar, basically we're marking updated Blocks in the for loops and using the smallest block_height to recompute later. I can update the code to track only the minimum block_height directly, but I kept it similar to the existing code. Performance benefit is likely pretty small, but could be updated.

The query deletes/overwrites all total counts higher than the smallest block_height block, since these would need to be updated relative to their parents/ancestors anyway.